bin/yk-config

```diff
@@ -56,9 +51,22 @@ handle_arg() {
             # Without this, clang will slap `optnone` attributes on every
             # function, causing optimisations to skip them.
             OUTPUT="${OUTPUT} -Xclang -disable-O0-optnone"
-            # XXX hacked in quick.
-            if [ -n "${PRELINK_PASSES}" ]; then
-                OUTPUT="${OUTPUT} -mllvm --newpm-passes=${PRELINK_PASSES}"
+            if [ ! -z "$2" ]; then
```

Comment:
I was expecting a top-level `--custom-prelink-pipeline` (or similarly named) which just overides a default pipeline.

In other words, there would be a top-level entry in the `case` construct for your new flag, which does nothing more than set a variable to override the pipeline. Then when the user provide `--cflags` the new pipeline is used.

Reply:
> wouldn't we need to inform the user ro pass this flag before --cflags

Yes, that was what I was expecting.

The sub-flag thing might be OK, although it does appear to complicate the implementation quite a bit.

Reply:
I actually don't know what line 56 does, so I'm not scared, but I am bamboozled ;)

That said, just because I'm stupid doesn't mean line 56 (or a minor variant using `&&`) isn't the right thing to do.

Reply:
It's a sub-parser for the `yk-config --cflags` switch, so you can have `yk-config --cflags -prelink BLAH` to specify a custom pipeline. The `--` check is to decide if the next switch is a top-level argument or a sub-argument to `--cflags`.

I had expected a top-level `--custom-prelink-pipeline` switch, which is simple to add, but does mean that it must come **before** `--cflags` on the command line (unless you jump through hoops).

Reply:
I'm happy with whatever the two of you decide is best.

Reply:
@vext01 I feel sub parser would be more intutive for user, and I can add a comment explaining what line 56 is doing,  please lmk what you think for this?

Reply:
What still bugs me a little is that there is no precedent for double hyphen for top-level args and single hyphen for sub-args. As a user I'd find this surprising.

FWIW, `yk-config`'s interface was based on `pkg-config` which has no sub-args.

If we really want sub-args, I think all args should be double-hyphen. That should be doable?

But honestly, if I were doing this, I'd have done the simpler implementation. I'll let you choose: simpler implementation or consistent parsing for sub-args.

Reply:
So I think the implementation wouldn't be much simpler, since we still need the string of passes after --prelink or --postlink. What I can do is add double hyphen for now. And see if I can implement both (pre and postlink) as args easily.

Reply:
Fixed here: [d2e295e](https://github.com/ykjit/yk/pull/829/commits/d2e295eef2863e9d606282d86616c8dc98200873)